examples.lisp

```diff
@@ -21,233 +21,142 @@
-
-      (let ((V (magicl::make-D-storage 4))
-            (D (magicl::make-D-storage 2))
-            (lwork 4096)
-            (liwork 4096)
-            (info 0)
-            (eigs-found 0))
-
-        (lapack:%dsyevr "V" "A" "U" 2 (magicl::copy-matrix-storage M) 2 0.0d0 0.0d0
-                        0 0 -1.0d0  eigs-found D V 2 (magicl::make-int32-storage 4)
-                        (magicl::make-D-storage lwork) lwork
-                        (magicl::make-int32-storage liwork) liwork
-                        info)
-        (format t "M = ~A~%V=~A~%D=~A~%~%" M V D)
-
-        ;; Construct a "matlab-style D" --- is there a better way?
-        (let ((Df (magicl::make-D-storage 4)))
-          (setf (aref Df 0) (aref D 0)
-                (aref Df 3) (aref D 1))
-          ;; Reconstruct M as V*Df*V';
-          (let ((Mri (magicl::make-D-storage 4))
-                (Mr (magicl::make-D-storage 4)))
-            (blas:%dgemm "N" "N" 2 2 2 1.0d0 V 2 Df 2 0.0d0 Mri 2)
-            (blas:%dgemm "N" "T" 2 2 2 1.0d0 Mri 2 V 2 0.0d0 Mr 2)
-            (format t "Reconstructed M = ~A~%" Mr)))))))
+  (let ((a (magicl:from-list '(1 2 3 4) '(2 2) :type 'double-float)))
+    (multiple-value-bind (val vect)
+        (magicl:eig a)
+      (format t "EVALUES = ~A~%EVECRTORS=~A~%~%" val vect))))
 
 (defun qr-example ()
-  (let ((a (make-complex-matrix 3 2 '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2) 4 #C(0 -2.9d0)))))
+  (let ((a (magicl:from-list '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2)) '(2 2) :type '(complex double-float))))
     (qr-printing a)))
 
 (defun ql-example ()
-  (let ((a (make-complex-matrix 3 2 '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2) 4 #C(0 -2.9d0)))))
+  (let ((a (magicl:from-list '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2)) '(2 2) :type '(complex double-float))))
     (ql-printing a)))
 
 (defun rq-example ()
-  (let ((a (make-complex-matrix 2 3 '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2) 4 #C(0 -2.9d0)))))
+  (let ((a (magicl:from-list '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2)) '(2 2) :type '(complex double-float))))
     (rq-printing a)))
 
 (defun lq-example ()
-  (let ((a (make-complex-matrix 2 3 '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2) 4 #C(0 -2.9d0)))))
+  (let ((a (magicl:from-list '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2)) '(2 2) :type '(complex double-float))))
     (lq-printing a)))
 
 (defun svd-example ()
-  (let ((a (make-complex-matrix 3 2 '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2) 4 #C(0 -2.9d0)))))
+  (let ((a (magicl:from-list '(#C(1 2) #C(-4 3) #C(-3 -3) #C(9 2)) '(2 2) :type '(complex double-float))))
     (svd-printing a)))
 
 (defun qr-printing (a)
   (multiple-value-bind (q r)
-      (qr a)
-    (let ((a-reconst (multiply-complex-matrices q r)))
-      (format t "A~%")
-      (print-matrix a)
-      (format t "Q~%")
-      (print-matrix q)
-      (format t "R~%")
-      (print-matrix r)
-      (format t "Reconstructed A~%")
-      (print-matrix a-reconst))))
+      (magicl:qr a)
+    (format t "A~%~a~%" a)
+    (format t "Q~%~a~%" q)
+    (format t "R~%~a~%" r)
```

Comment:
It's no big deal but you can merge all of these into a single `format`.

Reply:
I'm going to leave as is because it is consistent with the old ones (plus less work)